### PR TITLE
dcox chapter 5 exercises

### PR DESCRIPTION
**Pull request by codavex**

### CODE REVIEW

File: chapter05/dcox/ex5_2 [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+sub print_ruler_line {
+    print $_ % 10 foreach ( 1 .. 60 );
```
Comment (qianliok) [inline chapter05/dcox/ex5_2:7]: yeah, this looks much cooler. I'm curious to know what this will actually print out? so tempted to try...

Comment (pratyp) [inline chapter05/dcox/ex5_2:7]: Nice math! Putting all the chapters so far to good use i see :+1:

File: chapter05/dcox/ex5_1 [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+print reverse <> if (@ARGV);
```
Comment (pratyp) [inline chapter05/dcox/ex5_1:6]: Neat one-liner!

File: chapter05/dcox/ex5_3 [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# get one line
+chomp( my $column_width = <STDIN> );
+
+# default if rubbish is given
+$column_width = 60 if !looks_like_number($column_width);
+
+# get all the lines
+chomp( my @lines = <STDIN> );
+
+&print_ruler_line( $column_width + 10 );
```
Comment (pratyp) [inline chapter05/dcox/ex5_3:25]: Nice way to counter single digit inputs.

File: chapter05/dcox/ex5_2 (hunk 1 above)
Comment (qianliok) [inline chapter05/dcox/ex5_2:7]: Oh, I see. It's using math modulus operation to take the remainder. Great!

**Pull request merged**